### ISSUE TITLE
[HOLD #15352] [$1000] LHN - Account name is not displayed for a new chat in LHN

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#17150]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Prerequisites: Accounts A and B didn't have chat before
1. Login to NewDot with account A on any device and with account B on mobile (Android or iOS native apps)
2. Click "+" button under account A
3. Click New Chat
4. Input account B email and submit
5. Empty chat is displayed under account A
Optional step 6: send message from A to B

## Expected Result:
I would expect that chat on account B device should not appear until message is sent. Even if it should appear, then LHN should display name.

## Actual Result:
Chat appears under account B without sending any message. Account A name is not displayed under account B even after sending messages (you need to dive in to update name on LHN)

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [x] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.2.96.3

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/230673908-d189f259-8881-4f86-8e35-a1f5628031c5.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01786e3dda92e47708</li>
        <li>Upwork Job ID: 1645802614672261120</li>
        <li>Last Price Increase: 2023-05-02</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @anmurali (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
@anmurali Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01786e3dda92e47708

**Comment by MelvinBot** [comment]
Current assignee @anmurali is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @mananjadhav (`External`)

**Comment by hellohublot** [comment]
Maybe this is a backend problem, A clicks New Chat and selects B, the socket.report_xx.participants received by B is B himself, maybe it should be changed to A

**Comment by iwiznia** [comment]
@anmurali can you write down here the reasoning on why you thought this is external?

**Comment by anmurali** [comment]
I thought the account name not displaying for a new chat in LHN is a UI issue right? Purely front end? And only on Android presumably because in Chrome/ Mac the chat in the LHN does have the email.

>A clicks New Chat and selects B, the socket.report_xx.participants received by B is B himself, maybe it should be changed to A

This makes sense now that I try the same steps in Chrome where the email does show in LHN and I can see this above point illustrated

anu@expensify.com opens a chat with admin+04112023@tulacostarica.com
<img width="500" alt="image" src="https://user-images.githubusercontent.com/14225673/232176066-4fc3803b-0bdd-4791-82cf-bdf414e5db15.png">

That shows up to admin+04112023@tulacostarica.com as a chat from themselves....
To admin+04112023@tulacostarica.com it appears like below. See how the email of the sender is there in LHN
<img width="500" alt="image" src="https://user-images.githubusercontent.com/14225673/232176032-8b128107-fcfd-403b-9c54-f2ded2f88b12.png">

When I click into it, it resolves
<img width="500" alt="image" src="https://user-images.githubusercontent.com/14225673/232176234-33b24672-d34a-45ad-a173-3090c3ac8237.png">

Another issue is that the chat shows up at all for admin+04112023@tulacostarica.com as anu@expensify.com has simply created the DM but hasn't said anything yet. 

So my rationale was I thought the email not showing was a front end issue. But if this is all one big backend fix, I can remove that label and we can take it internal.

**Comment by iwiznia** [comment]
Oh, I have no idea, but since you added the label, I wanted to know the reasoning.
I agree that it is a bit weird that it happens on native and not on browsers and that probably indicates it is a frontend issue.

> A clicks New Chat and selects B, the socket.report_xx.participants received by B is B himself, maybe it should be changed to A

How does that explain that it works on browsers?

**Comment by MelvinBot** [comment]
@iwiznia @mananjadhav @anmurali this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by MelvinBot** [comment]
@iwiznia, @mananjadhav, @anmurali Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by MelvinBot** [comment]
@iwiznia, @mananjadhav, @anmurali 6 days overdue. This is scarier than being forced to listen to Vogon poetry!

**Comment by iwiznia** [comment]
cc @mananjadhav

**Comment by mountiny** [comment]
@iwiznia @mananjadhav this should be on hold for this project https://github.com/Expensify/App/issues/15352, then the created chats with no messages should not show up to users.

**Comment by alitoshmatov** [comment]
Related https://github.com/Expensify/App/issues/17429

**Comment by anmurali** [comment]
@mountiny - should I close the Upwork Job and put this GH on [Hold] in favor of getting https://github.com/Expensify/App/issues/15352 done first?

**Comment by MelvinBot** [comment]
📣 It's been a week! Do we have any satisfactory proposals yet? Do we need to adjust the bounty for this issue? 💸

**Comment by mountiny** [comment]
Updated this

**Comment by iwiznia** [comment]
Not overdue since it is on hold

**Comment by anmurali** [comment]
Let's also make it a Weekly

**Comment by iwiznia** [comment]
Still on hold.

**Comment by iwiznia** [comment]
Same

**Comment by anmurali** [comment]
still on hold waiting on #15352

**Comment by anmurali** [comment]
https://github.com/Expensify/App/issues/15352#issuecomment-1622530934
So we closed the issue this was held on as no longer needed. Does this mean we can pick this up now? @mountiny @iwiznia

**Comment by mountiny** [comment]
I think we are going to be taking a different approach to solve the same issue, is this still reproducible though @mananjadhav ?

**Comment by anmurali** [comment]
>I would expect that chat on account B device should not appear until message is sent. 

This is for sure still happening
>Even if it should appear, then LHN should display name.

This might be fixed.

**Comment by iwiznia** [comment]
Going on sabbatical, please reassign another engineer

**Comment by mananjadhav** [comment]
I tried reproducing this on old and new accounts both. I am unable to reproduce this.  @kbecciv Are you still able to reproduce this?

@anmurali @mountiny Best to close this one out?

**Comment by mountiny** [comment]
Yep lets close this as it was  most likely solved with the secure login migration
